internal/telemetry/tracker_test.go

```diff
@@ -0,0 +1,103 @@
+package telemetry
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+type testClient struct {
+	calls []interface{}
+}
+
+func (client *testClient) Enqueue(track analytics.Message) error {
+	client.calls = append(client.calls, track)
+	return nil
+}
+
+func (client *testClient) Close() error {
+	return nil
+}
+
+const (
+	testID          = "id123"
+	testUserID      = "user123"
+	testExecutionID = "execution123"
+)
+
+var testTime = time.Date(2021, 1, 2, 3, 4, 5, 6, time.UTC)
+
+func TestNoopTracker(t *testing.T) {
+	t.Run("should create a noop tracker and should not do anything when track is called", func(t *testing.T) {
+		tracker := noopTracker{}
+		testEvent := createEvent(EventTypeCommandStart, nil, "someCommand")
+		testTrackStdoutOutput(t, &tracker, testEvent, "")
+	})
+}
+
+func TestStdoutTracker(t *testing.T) {
+	t.Run("should create an stdout tracker and should print the tracking information to stdout", func(t *testing.T) {
+		tracker := stdoutTracker{}
+		testTrackStdoutOutput(
+			t,
+			&tracker,
+			createEvent(EventTypeCommandComplete, nil, "someCommand"),
+			"03:04:05 UTC TELEM someCommand: COMMAND_COMPLETE[]\n",
+		)
+	})
+}
+
+func TestSegmentTracker(t *testing.T) {
+	client := &testClient{}
+	t.Run("should create the segment tracker and should print the tracking information to the logger", func(t *testing.T) {
+		tracker := segmentTracker{}
+		tracker.client = client
+		tracker.Track(createEvent(EventTypeCommandError, []EventData{{Key: EventDataKeyErr, Value: "Something"}}, "someCommand"))
+		testClientResults := tracker.client.(*testClient)
+		assert.Equal(t, 1, len(testClientResults.calls))
+
+		expectedTrack := analytics.Track{
+			MessageId: testID,
+			UserId:    testUserID,
+			Timestamp: testTime,
+			Event:     string(EventTypeCommandError),
+			Properties: map[string]interface{}{
+				string(EventDataKeyErr): "Something",
+			},
+		}
+		actualTrack := testClientResults.calls[0].(analytics.Track)
```

Comment:
out of curiosity, was something failing when you tried to run:

```go
assert.Equal(t, []interface{}{expectedTrack}, testClientResults.calls)
```

If something like that works let's use it, I'd like to keep the asserts as simple as possible.

If I'm missing something there and that doesn't work, let's at least avoid any potential panics within this test.  You should always try to use the two-return-value `, ok` syntax in tests and `assert.True` on the `ok` var (this gives us a nicer test failure message than panics will).  Also, the last assert here ("equals" on the two tracks) should cover the two asserts before that ("non nil" and "length of properties is 1")